modules/commons/src/main/frontend/src/components/DropdownsDatePicker.jsx

```diff
@@ -0,0 +1,295 @@
+          {monthShort ? moment().month(i).format("MMM") : moment().month(i).format("MMMM")}
+        </MenuItem>
+      );
+    }
+
+    return monthOptions;
+  }
+
+  let generateDayOptions = () => {
+    let dayOptions = [];
+    dayOptions.push(<MenuItem key={-1} value="-1">dd</MenuItem>);
+
+    let start = (selectedYear === startYear && selectedMonth === startMonth) ? startDay : 1;
+    const monthDays = getDaysInMonth(selectedYear, selectedMonth);
+    let end = (selectedYear === endYear && selectedMonth === endMonth) ? endDay : monthDays;
+
+    for (let i = start; i <= end; i++) {
+      dayOptions.push(<MenuItem key={i} value={i}>{i}</MenuItem>);
+    }
+
+    return dayOptions;
+  }
+
+  let handleDateChange = (type, value) => {
+    let dateObj = {
+      [DropdownDate.year] : (type === DropdownDate.year) ? value : selectedYear,
+      [DropdownDate.month] : (type === DropdownDate.month) ? value : selectedMonth,
+      [DropdownDate.day] : (type === DropdownDate.day) ? value : selectedDay
+    };
+
+    if (dateObj[DropdownDate.year] !== -1 && dateObj[DropdownDate.month] !== -1 && dateObj[DropdownDate.day] !== -1) {
+      let date = new Date(dateObj[DropdownDate.year], dateObj[DropdownDate.month], dateObj[DropdownDate.day]);
+
+      if (formatDate) {
+        dateObj[DropdownDate.month] = dateObj[DropdownDate.month] + 1;
+        if (dateObj[DropdownDate.month] < 10) dateObj[DropdownDate.month] = '0' + dateObj[DropdownDate.month];
+        if (dateObj[DropdownDate.day] < 10) dateObj[DropdownDate.day] = '0' + dateObj[DropdownDate.day];
+        date = order.map(part => { return dateObj[part] }).join('-');
+      }
+
+      onDateChange(date);
+    }
+  }
+
+  let handleYearChange = (event) => {
+    const year = parseInt(event.target.value);
+    setSelectedYear(year);
+    const monthDays = getDaysInMonth(year, selectedMonth);
+    if (selectedDay > monthDays) {
+      setSelectedDay(-1);
+    } else {
+      onDateChange && handleDateChange(DropdownDate.year, year);
+    }
+  }
+
+  let handleMonthChange = (event) => {
+    const month = parseInt(event.target.value);
+    setSelectedMonth(month);
+    const monthDays = getDaysInMonth(selectedYear, month);
+    if (selectedDay > monthDays) {
+      setSelectedDay(-1);
+    } else {
+      onDateChange && handleDateChange(DropdownDate.month, month);
+    }
+  }
+
+  let handleDayChange = (event) => {
+    const day = parseInt(event.target.value);
+    setSelectedDay(day);
+    onDateChange && handleDateChange(DropdownDate.day, day);
+  }
+
+  let renderYear = () => {
+    return (
+      <Select
+        key="year"
+        onChange={handleYearChange}
+        value={selectedYear}
+        disabled={disabled}
+        className={order.indexOf("year") != 0 ? classes.select : ''}
+        autoFocus={autoFocus && order.indexOf("year") == 0}
+        renderValue={(selected) => {
+            if (selected < 0 ) {
+              return <div className={classes.default}>yyyy</div>;
```

Comment:
Now that we're not tied to the native date picker, let's make the components more user friendly with human-readable placeholders.

```suggestion
              return <div className={classes.default}>Year</div>;
```